bluesky/run_engine.py

```diff
@@ -734,7 +734,16 @@ def _read(self, msg):
         obj = msg.obj
         self._objs_read.append(obj)
         if obj not in self._describe_cache:
-            self._describe_cache[obj] = obj.describe()
+            # Validate that there is no data key name collision.
+            data_keys = obj.describe()
+            keys = data_keys.keys()  # that is, field names
+            for known_obj, known_data_keys in self._describe_cache.items():
```

Comment:
This is a _bit_ more restrictive than we need it to be, we should be able to use objects with conflicting key names in the same scan as long as we don't bundle them into the same event.  On the other hand, we will deal with that complication when we find a beamline that needs it.

I like that this is a one time check.